libraries/common/helpers/redux/shouldFetchData.js

```diff
@@ -26,20 +26,20 @@ export function shouldFetchData(item, itemKey = null, requiredCount = null) {
       return true;
     }
 
-    // Check for parameters that enable sub item counting
-    const checkSubItems = typeof item.totalResultCount !== 'undefined' || requiredCount !== null;
-
     // Check for a specific key inside items and fetch it if it's not found.
-    if (checkSubItems && itemKey && Array.isArray(item[itemKey])) {
-      let maxItems = requiredCount || item.totalResultCount;
-
-      // Check if we are requesting more items that there is available.
-      if (maxItems > item.totalResultCount) {
-        maxItems = item.totalResultCount;
+    if (itemKey && Array.isArray(item[itemKey])) {
+      // 1. Fetch by requiredCount.
+      if (requiredCount) {
+        const assertCount = Math.min(requiredCount, item.totalResultCount);
```

Comment:
This line will fail if `totalResultCount` is not set. Since this is a general helper, we can't make assumptions about the item structure.